client/src/App.tsx

```diff
@@ -29,30 +25,37 @@ function App(): JSX.Element {
       <ThemeProvider theme={theme}>
         <QueryClientProvider client={queryClient}>
           <ReactQueryDevtools />
-          <UserProvider>
-            <ModalProvider>
-              <BrowserRouter
-                getUserConfirmation={() => {
-                  // intentionally left empty callback to block the default browser prompt.
-                }}
-              >
-                <Grid
-                  className="App"
-                  display="flex"
-                  flexDirection="column"
-                  height="100vh"
-                  width="100vw"
-                  justifyContent="space-between"
+          <ErrorBoundary
+            FallbackComponent={ErrorFallback}
+            onReset={() => {
+              /* reset logic here */
+            }}
+          >
+            <UserProvider>
+              <ModalProvider>
+                <BrowserRouter
+                  getUserConfirmation={() => {
+                    // intentionally left empty callback to block the default browser prompt.
+                  }}
                 >
-                  {/* move modal to below footer after troubleshooting */}
-                  <Modal />
-                  <Header />
-                  <Main />
-                  <Footer />
-                </Grid>
-              </BrowserRouter>
-            </ModalProvider>
-          </UserProvider>
+                  <Grid
+                    className="App"
+                    display="flex"
+                    flexDirection="column"
+                    height="100vh"
+                    width="100vw"
+                    justifyContent="space-between"
+                  >
+                    {/* move modal to below footer after troubleshooting */}
+                    <Modal />
+                    <Header />
+                    <Main />
```

Comment:
It might also be good to wrap `Main` around with an `ErrorBoundary`.  This is so when an error is inside the main component, we will still have a header and footer.

<img width="1461" alt="Screenshot 2023-09-11 at 6 25 28 AM" src="https://github.com/Nonprofit-Exchange-Hub/web-app/assets/2562488/1ae1e752-acca-43d5-986b-91c1b7ab01eb">

Reply:
Let's reach out to design team and see if they can help create a display for these kinds of errors (including 404 - page not found) .. maybe they can have some fun with it like in some of these https://www.creativebloq.com/web-design/best-404-pages-812505 ... which reminds me, I need to do the same with the loading and error states for the user sign-up form (will drop a message on the design channel later today).

Reply:
404 Pages are always the tell if the devs cared or not (or if they had plenty of time or were rushed perhaps?). Either way i like the idea!

Moving the ErrorBoundary around the Main makes sense for now unless we need a greater scope, which should only require a simple move once again.